var/spack/repos/builtin/packages/legion/package.py

```diff
@@ -39,24 +39,46 @@ class Legion(CMakePackage):
 
     variant('mpi', default=True,
             description='Build on top of mpi conduit for mpi inoperability')
+    variant('ibv', default=False,
+            description='Build on top of ibv conduit for InfiniBand support')
     variant('shared', default=True, description='Build shared libraries')
     variant('hdf5', default=True, description='Enable HDF5 support')
+    variant('build_type', default='Release', values=('Debug', 'Release'),
+            description='The build type to build')
 
     depends_on("cmake@3.1:", type='build')
     depends_on("gasnet~aligned-segments~pshm segment-mmap-max='16GB'", when='~mpi')
     depends_on("gasnet~aligned-segments~pshm segment-mmap-max='16GB' +mpi", when='+mpi')
+    depends_on("gasnet~aligned-segments~pshm segment-mmap-max='16GB' +ibv", when='+ibv')
     depends_on("hdf5~mpi", when='+hdf5')
 
     def cmake_args(self):
+        cmake_cxx_flags = [
+            '-DPRIVILEGE_CHECKS',
+            '-DBOUNDS_CHECKS',
+            '-DENABLE_LEGION_TLS']
+
         options = [
             '-DLegion_USE_GASNet=ON',
+            '-DLEGION_USE_CUDA=OFF',
+            '-DLEGION_USE_OPENMP=OFF',
             '-DLegion_BUILD_EXAMPLES=ON',
             '-DBUILD_SHARED_LIBS=%s' % ('+shared' in self.spec)]
 
+        if self.spec.variants['build_type'].value == 'Debug':
+            options.append('-DCMAKE_BUILD_TYPE=Debug')
```

Comment:
@scheibelp doesn’t that get added automatically already.

Reply:
Yes `-DCMAKE_BUILD_TYPE=Debug` should be added automatically even when it is overridden as it is here (if that's not happening it's an issue with Spack)

The CXX flags (on the next line) would not be added automatically

Reply:
Is the automatically added value still check under `self.spec.variants['build_type']`?